app/src/org/commcare/views/dialogs/CommCareAlertDialog.java

```diff
@@ -19,16 +19,20 @@
     protected DialogInterface.OnCancelListener cancelListener;
     private DialogInterface.OnDismissListener dismissListener;
     protected View view;
-    // false by default, can be overridden by calling setOnCancelListener(), or by subclass
-    // definitions if desired
+    // false by default, can be overridden if desired
     protected boolean isCancelable;
 
     public void finalizeView() {
         dialog.setCancelable(isCancelable);
-        if (isCancelable) {
+        if (cancelListener != null) {
```

Comment:
seems slightly redundant having both cancelListener and isCalcelable when you're checking cancelListener != null - is there a case where the dialog is cancelable but there isn't a listener?